src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1925,7 +1926,8 @@ private ITypeSymbol UnwrapTaskLike(ITypeSymbol type, bool isAsync)
                 {
                     if (type.OriginalDefinition.Equals(this.Compilation.TaskOfTType()))
                     {
-                        return ((INamedTypeSymbol)type).TypeArguments[0];
+                        var namedTypeSymbol = (INamedTypeSymbol)type;
+                        return namedTypeSymbol.TypeArguments[0].WithNullability(namedTypeSymbol.TypeArgumentsNullableAnnotations[0]);
```

Comment:
maybe extensions for this as well?  basically, these sorts of repetitive patterns to combine the data might benefit from easy to access convenience methods that just do that for you.  i.e. `namedTypeSymbol.GetTypeArgumentWithNullability(0)`